### PR TITLE
add count to runs feed gql

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation
Updates the Graphene layer to expose a `count` attr on `RunsFeedOrError`. Then uses it in the UI to display counts for the Queued and In Progress tabs

## How I Tested These Changes
On All Runs tab, can see the In Progress count
Screenshot 2024-09-25 at 3 16 17 PM

In Progress tab correctly filters
Screenshot 2024-09-25 at 3 16 05 PM

#### With a filter applied
On All Runs tab, can see the In Progress count
Screenshot 2024-09-25 at 3 19 26 PM

In Progress tab correctly filters
Screenshot 2024-09-25 at 3 19 31 PM

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by jamiedemaria** [comment]
* **#24745** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24745?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jamiedemaria and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py [hunk 1]
```diff
@@ -842,13 +841,16 @@ async def resolve_runOrError(self, graphene_info: ResolveInfo, runId):
     def resolve_runsFeedOrError(
         self,
         graphene_info: ResolveInfo,
-        limit: int,
+        limit: Optional[int] = None,
```
Comment (OwenKephart) [inline python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:844]: Why make this optional?

**Review by OwenKephart — COMMENTED** [review]
Looks reasonable, but note comment -- typically we've been trying to move to enforcing limits across queries to avoid bad behaviors.

My understanding is that this is optional because the limit argument doesn't really make sense if you're getting the count (as we don't apply any limit in that case, nor should we). However, I wonder if that weirdness indicates that we should put this field on an object other than GrapheneRunsFeed, and instead have a special-purpose count object

File: python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py (hunk 1 above)
Comment (jamiedemaria) [reply python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:844]: yeah, your idea is right. basically it didn't make sense to have limit (or cursor, but that was already optional) for getting the count. but it also felt gross the make the limit optional and have a runtime error if it's not passed to the resolver. idk what the right pattern to do here is or if we've had similar situations in the past to reference

Comment (jamiedemaria) [reply python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:844]: a separate object is appealing though

Comment (OwenKephart) [reply python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:844]: given the weirdness introduced, I think I would prefer a solution where this count is a separate query -- I believe you can bundle together multiple queries into the same request, so having these be independent feels like the right call, e.g.

```
query RunsFeedEntryQuery($cursor: String, $limit: Int!, $filter: RunsFilter) {
  runsFeedOrError(cursor: $cursor, limit: $limit, filter: $filter) { ... }
  runsCountOrError(filter: $filter) {...}
}
```

Comment (jamiedemaria) [reply python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py:844]: yeah i went ahead and started it here https://github.com/dagster-io/dagster/pull/24813

**Comment by jamiedemaria** [comment]
closing in favor of https://github.com/dagster-io/dagster/pull/24813 where it's implemented as a separate query

**Pull request closed without merging**